apache/couchdb-fauxton
# Replace brace with react-ace

## Overview

Replaces the `brace` dependency, which hasn't received any updates for 5+ years, with `react-ace` which works with the  most recent releases of the ACE Editor. The work was based on https://github.com/apache/couchdb-fauxton/pull/1331 which had to be reverted due to issues with the release build. Both the dev server and release builds are fully functional.

Also takes advantage of the new `useStrictCSP` option in the ACE Editor so allow Fauxton to work without the `style-src: 'unsafe-inline'` Content-Security Policy.
 
## Testing recommendations

When editing JSON documents:
- Syntax highlighting is working
- Editor shows syntax errors - e.g. remove a comma between fields and the red X icon should be displayed on the editor gutter
- String editor:
  - position the cursor inside a string value then click the "String Editor" icon displayed on the editor gutter
  - In the String Editor, modify the string value then click "Modify text" button. 
  - Verify the string value is updated

When editing a Mango Query:
- Syntax highlighting is working
- Editor shows syntax errors
- Click the "Enter zen mode" icon on the top right of the editor to enter the full-screen editor
  - Click the "Switch zen theme" icon on the right to verify it switches between the light / dark themes

## GitHub issue number

n/a

## Related Pull Requests

n/a

## Checklist

- [ ] Update [rebar.config.script](https://github.com/apache/couchdb/blob/main/rebar.config.script) with the correct tag once a new Fauxton release is made
